AxonDeepSeg/visualization/visualize.py

```diff
@@ -1,174 +1,218 @@
-import pickle
+import os
 import sys
-from sys import platform as _platform
-if 'pytest' in sys.modules:
-    import matplotlib as mpl
-    mpl.use('Agg') # Enforces mpl to not open new plot windows
-elif _platform == "darwin": # Mac OSX
-    import matplotlib as mpl
-    mpl.use('TkAgg')
-import matplotlib.pyplot as plt
-from scipy.misc import imread
+import pickle
+
+# Scientific modules imports
+import numpy as np
 from sklearn.metrics import accuracy_score
-from ..testing.segmentation_scoring import score_analysis, dice
 from sklearn import preprocessing
-import os
+from scipy.misc import imread
 from tabulate import tabulate
-import numpy as np
+
+# Graphs and plots imports
+from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+from matplotlib.figure import Figure
+
+# AxonDeepSeg imports
 import AxonDeepSeg.ads_utils
+from ..testing.segmentation_scoring import score_analysis, dice
 
 
-def visualize_training(path_model, start_visu=0):
+def visualize_training(path_model, start_for_viz=0):
     """
     :param path_model: path of the folder with the model parameters .ckpt
-    :param start_visu: first iterations can reach extreme values, start_visu set another start than epoch 0
-    :return: evolution
-
-    figure(1) represent the evolution of the loss and the accuracy evaluated on the validation set along the learning process.
-    If the learning began from an initial model, the figure plots first the accuracy and loss evolution from this initial model and then stacks the evolution of the model.
+    :param start_visu: first iterations can reach extreme values, 
```

Comment:
@maf88 if we do indeed keep the change to `start_for_viz`, this line and the line below it should be changed accordingly.